RocketDataTests/CollectionDataProviderTests/ConsistencyCollectionDataProviderTests.swift

```diff
@@ -229,6 +229,77 @@ class ConsistencyCollectionDataProviderTests: RocketDataTestCase {
         waitForExpectationsWithTimeout(10, handler: nil)
     }
 
+    /**
+     If we insert an item into a collection, possibly, other models in the collection should update.
+     So in this test, all the models have a submodel. We insert a new model which has an updated submodel.
+     This submodel should get updated everywhere in this collection.
+     
+     Specifically, we have two models initially. One has a submodel with id 2, the other without an id.
+     We add a new model to the array which has an updated submodel of id 2.
+     We expect the consistency manager to update this row later.
+     */
+    func testInsertUpdatesOtherItemsInCollections() {
+        let dataProvider = CollectionDataProvider<ParentModel>(dataModelManager: DataModelManager.sharedDataManagerNoCache)
+        let otherDataProvider = CollectionDataProvider<ParentModel>(dataModelManager: DataModelManager.sharedDataManagerNoCache)
+
+        let initialChildModel = ChildModel(id: 2, name: "initial")
+        let initialModel = ParentModel(id: 1, name: "initial", requiredChild: initialChildModel, otherChildren: [])
+        let otherModel = ParentModel(id: 3, name: "other", requiredChild: ChildModel(), otherChildren: [])
+        let newChild = ChildModel(id: 2, name: "new")
+        let newModel = ParentModel(id: 5, name: "new", requiredChild: newChild, otherChildren: [])
+
+        var delegateCalled = 0
+        let delegate = ClosureCollectionDataProviderDelegate() { (collectionChanges, context) in
+            delegateCalled += 1
+            switch delegateCalled {
+            case 1:
+                XCTAssertEqual(context as? String, "context")
+                XCTAssertEqual(dataProvider[2].name, "new")
+                XCTAssertEqual(collectionChanges.count, 1)
+                XCTAssertEqual(collectionChanges[0], CollectionChangeInformation.insert(index: 2))
+            case 2:
```

Comment:
Does case 1 happen synchronously and case 2 asynchronously?
It seems slightly odd that the original collection doesn't get a both changes simultaneously.